### PR TITLE
UNS domain support

### PR DESCRIPTION
**Pull request by mul1sh**
Replaces #1010 

Will fix https://github.com/unstoppabledomains/gitcoin-bounties/issues/3

### CODE REVIEW

**Comment by mhluongo** [comment]
@mul1sh I haven't reviewed this as it's in draft — is it still in progress?

**Comment by 0xJordi** [comment]
Hi Matt @mhluongo! I'm Jordi from the UD team.

@mul1sh is working on it and should be live soon! I'm also working with Chloé to announce this :)

**Comment by mul1sh** [comment]
> @mul1sh I haven't reviewed this as it's in draft — is it still in progress?

Yes ser, just fixing some merge conflicts locally and it will be ready to review later today.

**Comment by 7alip** [comment]
Hey @mul1sh

There seems to be an important refactor on NS. You may want to consider checking #1304 before continuing your implementation. It took my attention as I've been considering to add [RNS](https://www.rifos.org/nameservice/)

**Comment by mul1sh** [comment]
> Hey @mul1sh
> 
> There seems to be an important refactor on NS. You may want to consider checking #1304 before continuing your implementation. It took my attention as I've been considering to add [RNS](https://www.rifos.org/nameservice/)

Ahh ok thanks for the 

> Hey @mul1sh
> 
> There seems to be an important refactor on NS. You may want to consider checking #1304 before continuing your implementation. It took my attention as I've been considering to add [RNS](https://www.rifos.org/nameservice/)

Ah ok, thanks for the heads up will look into it asap! 🙂

**Comment by mul1sh** [comment]
> Hey @mul1sh
> 
> There seems to be an important refactor on NS. You may want to consider checking #1304 before continuing your implementation. It took my attention as I've been considering to add [RNS](https://www.rifos.org/nameservice/)

Ah ok thanks for the heads up, will look into it asap! 🙂

**Comment by Shadowfiend** [comment]
Oof, sorry for not flagging here, I finished this right before going on vacation and it got merged while I was out so I didn't get a chance to post the note I wanted! @7alip thank you for calling it out 🙏

File: background/services/name/resolvers/uns.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      return {}
+    },
+    async lookUpNameForAddress({
+      address,
+      network,
+    }: AddressOnNetwork): Promise<NameOnNetwork | undefined> {
+      return {
+        name: "",
+        network: undefined,
+      }
```
Comment (Shadowfiend) [inline background/services/name/resolvers/uns.ts:81]: For these two, if unsupported, the async function should throw an exception (see e.g. [the known contract resolver's `lookUpAvatar` implementation](https://github.com/tallycash/extension/blob/main/background/services/name/resolvers/known-contracts.ts#L22-L24)).

Similarly, if UNS doesn't support reverse resolution, `canAttemptNameResolution` should return `false`.

See https://github.com/tallycash/extension/blob/main/background/services/name/name-resolver.ts for full API docs!

**Comment by mul1sh** [comment]
> Oof, sorry for not flagging here, I finished this right before going on vacation and it got merged while I was out so I didn't get a chance to post the note I wanted! @7alip thank you for calling it out 🙏

No worries at all, I understand :) I'm still working on the PR, so I'll push a commit which fixes the issues noted above in a few hours.

**Comment by mul1sh** [comment]
@mhluongo @Shadowfiend @0xJordi  This is now finally ready for review, thanks for the patience as I worked on it 🎉 💯 .

Though now it seems the domain resolution is broken when sending funds to a domain name. This is the case even for ENS which was working before, what could be the issue? 🤔

**Comment by mul1sh** [comment]
> @mhluongo @Shadowfiend @0xJordi This is now finally ready for review, thanks for the patience as I worked on it 🎉 💯 .
> 
> Though now it seems the domain resolution is broken when sending funds to a domain name. This is the case even for ENS which was working before, what could be the issue? 🤔

Sorry, upon further investigation it seems this issue is only with this branch i.e. `uns-ens-domain-support`, branching out from main and applying the same changes in this PR doesn't bring any issues at all. ENS & UNS are resolving as required in all the necessary places in the UI.

So once again I will have to close this PR and open a new one from the main branch and then PR my changes from there.

**Pull request closed without merging**